### ISSUE TITLE
Add teaching and talks

### ISSUE THREAD

**Issue by apotiron** [jsonresume/resume-schema#466]
Hi everyone,

Thanks for developing and maintaining this project. I stumbled upon it recently and I found it very cool. Yet, I struggle to see how to add teaching experience as an academic, e.g., course, semesters, description, etc. I have read the issue #305 and thought to use the class projects but then, I saw the last commit (1ab9095) saying that no one uses it. So what the status @thomasdavis ? How do you add your teaching experience then?

I have a very similar comment regarding the talks. I have read the issue #308 and it seems that the class projects is also used but then again, the last commit (1ab9095) seems to remove it. How do you add the talks you give in conferences? I was thinking maybe using a version of the class publications?

Thank you!
All the best

**Comment by tgrushka** [comment]
Yes, would really like to see this under something like "talks" or "presentations" as Conference Presentations is a very common section, especially for academics or hybrid pro-academics. These are not always associated with a publication; in fact, I had a paper accepted for presentation but not publication at a conference. I will use "projects" for now (I see "projects" has a "type" which can be "conference", "talk", or "presentation") but it seems a bit hacky, as one "project" could be presented at several conferences. Also, often (in my case) these are listed along side publications (i.e. "Publications and Conference Presentations"), especially if one has fewer publications and is trying to save space, so would be nice if these sections could be "compatible" and combined in a theme if the user wishes.

**Comment by tgrushka** [comment]
Hi @thomasdavis , I feel an urgency for this issue because I need to get my resume out ASAP and might just have to fork the schema to my own to customize it, which I don't want to do because it deviates from and waters down the standard you've been establishing.

Hope you might be willing to expand the academic abilities of this schema a bit as this issue is asking. Like you, I'm a software engineer -- https://registry.jsonresume.org/tgrushka -- but also part academic and have numerous conference presentations and a couple of publications to my name, and a very eclectic background, so I might be one of your worst test cases for the schema's flexibility! But the problem with using projects for conference presentations is like trying to fit a square peg into a round hole. Additionally, presentations are very different from either publications or projects.

1. Is it possible to add an "author" or "authors" key to publication entries?

  * Publications need an "author" or "authors" field. Published papers/articles sometimes, but most often not, have only a single author. But the current schema does not even have a field for a single author -- assuming the resume "owner" is the author, or even the first one. One can contribute many ways to publications, and because it's so competitive, surely being a second, third, or even a fourth author (like me) carries weight.

2. May we please have some support in the near future for presentations?

  * These might have fields such as (draft idea only):
    * **title** - the title of the presentation
    * **authors** - optional/nullable - the person or people who authored the project/study/publication being presented
    * presenters (maybe?) - optional/nullable - the person or people actually presenting - maybe authors would be sufficient though, as authors should always be listed - e.g. I was a "presenter" at a conference in Honolulu in absentia - my professor included me because I helped with the study, so maybe authors is sufficient, but some may disagree
    * **location** - optional/nullable - the geographic location, city, region, country, etc.
    * **event** or **venue** - title of the whole conference, exhibition, meeting, event, etc.
    * **type** - could be conference, meeting, symposium, exhibition, etc.
    * **date** - date given on
    * **highlights** - like other highlights sections

  * I'm not completely attached to the name "presentations", but am a bit opposed to "talks" or even "conferences", because presentations could also include poster presentations, art, music, workshops, etc. that are presented in or out of a "conference" venue setting -- perhaps at symposia, which some consider to be different then conferences, or museums, art installations or shows, etc., so what category would the others fall under? I do feel that "presentations" seems to be pretty broad, and that "projects" could certainly catch things that are not "presentations", but someone else may have a different opinion.

  * It seems **the category key name is important for AI optimization**, as I have also experienced issues ingesting my JSON Resume into ChatGPT in other areas due to strange names, (i.e. "name" under "work" for the employer, instead of "company" or "employer" -- is "name" the "title" of the role or the "company"?). I'm even working on a transformation script to get my JSON Resume into a more friendly ingestion format for ChatGPT so I can automate it. This is becoming more and more critical nowadays for remaining competitive in the labor market and even "AIO" - Artificial Intelligence Optimization, like Search Engine Optimization - as more and more AI is being used to filter candidates, and we need to make sure our resumes are really fine-tuned to each position just to get through the first recruiting round!
  * **Conference presentations are not publications, in any way, shape, or form.** A person's work can be accepted for one but not the other. Presentations can also be much less formal, i.e. workshops, or presentations that would be unsuitable for publication. It could even hurt the credibility of someone to list a presentation under "Publications" or even "Projects" (because the reviewer of the CV/resume may not realize it was done as a technical workaround for the resume format as opposed to a false claim -- especially for academic posts like professorships or researchers).
  * Often, one might have **several presentations of one "project"** -- i.e. a research project. It would be a **nice-to-have**, but certainly less important than the basic fields needed, to be able to reference a project from both a publication and a presentation.

3. I definitely upvote the teaching section, although I don't have any yet, but would like to do so. This could include anything from university, college, or even high school teaching, to giving bootcamps, to YouTube or other tutorials, or workshops, depending on the candidate and their career goals.

4. Thank you for keeping "projects". It's still very useful even with the other sections added. Also, the sections could be optional (I believe they are already?) so it can't hurt to have them. I agree simplicity / less ambiguity is better though, otherwise the data organization/search/retrieval becomes less effective.

5. Maybe a separate issue -- would be nice if the "work" / "projects" "endDate" could be optional or null to indicate that a role is current / present / ongoing. Maybe it is. VS Code complains when I assign null to it, though, so maybe the schema could include this, or is that a tech limitation of JSON schemas?

Thanks so much, your schema is awesome and so much better than anything else out there for keeping a master of one's resume data. With just a few tweaks, it could be a bit more versatile for more professions including academic.
